### PR TITLE
Base/push

### PR DESCRIPTION
**Pull request by MehrdadMzfr**

### CODE REVIEW

File: MobileCenterPush/MobileCenterPush/MSPush.m [hunk 1]
```diff
@@ -102,50 +103,45 @@ + (void)resetSharedInstance {
...
-    UNAuthorizationOptions authOptions =
-        (UNAuthorizationOptions)(UNAuthorizationOptionAlert | UNAuthorizationOptionSound | UNAuthorizationOptionBadge);
+    UNAuthorizationOptions authOptions = (UNAuthorizationOptions) (UNAuthorizationOptionAlert | UNAuthorizationOptionSound | UNAuthorizationOptionBadge);
     [center requestAuthorizationWithOptions:authOptions
-                          completionHandler:^(__attribute__((unused)) BOOL granted,
-                                              __attribute__((unused)) NSError *_Nullable error){
-                          }];
+                          completionHandler:^(BOOL granted, NSError * _Nullable error) {}];
+    
+    [[UNUserNotificationCenter currentNotificationCenter] setDelegate:[[UIApplication sharedApplication] delegate]];
```
Comment (clpolet) [inline MobileCenterPush/MobileCenterPush/MSPush.m]: Do we actually need this as of now? Looks like we don't really need it except if we decide to override the OS behavior and show system notifications while app in foreground for instance? If so we should provide our own delegate for people that doesn't want to bother implementing it. See ME for examples.

File: MobileCenterPush/MobileCenterPush.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -403,7 +403,7 @@
...
..._DIR}/Release-iphonesimulator\n\n# Cleaning previous build\nxcodebuild -project \"${FMK_NAME}.xcodeproj\" -configuration \"Release\" -target \"${FMK_NAME}\" clean\n\n# Building both architectures.\nxcodebuild -project \"${FMK_NAME}.xcodeproj\" -configuration \"Release\" -target \"${FMK_NAME}\" -sdk iphoneos\nxcodebuild -project \"${FMK_NAME}.xcodeproj\" -configuration \"Release\" -target \"${FMK_NAME}\" -sdk iphonesimulator\n\n# Cleaning the oldest.\nif [ -d \"${INSTALL_DIR}\" ]\nthen\nrm -rf \"${INSTALL_DIR}\"\nfi\n\n# Creates and renews the final product folder.\nmkdir -p \"${INSTALL_DIR}\"\nmkdir -p \"${INSTALL_DIR}/Headers\"\nmkdir -p \"${INSTALL_DIR}/Modules\"\n\n# Copy the swift import file\ncp -f \"${SRCROOT}/${FMK_NAME}/Support/module.modulemap\" \"${INSTALL_DIR}/Modules/\"\n\n# Copies the headers and resources files to the final product folder.\ncp -R \"${SRCROOT}/${WRK_DIR}/Release-iphoneos/include/${FMK_NAME}/\" \"${INSTALL_DIR}/Headers/\"\n\n\n# Uses the Lipo Tool to merge both binary files (i386 + armv6/armv7) into one Universal final product.\nlipo -create \"${DEVICE_DIR}/lib${FMK_NAME}.a\" \"${SIMULATOR_DIR}/lib${FMK_NAME}.a\" -output \"${INSTALL_DIR}/${FMK_NAME}\"";
```
Comment (MehrdadMzfr) [inline MobileCenterPush/MobileCenterPush.xcodeproj/project.pbxproj]: resolved

File: MobileCenterPush/MobileCenterPush/MSPush.m [hunk 3]
```diff
@@ -42,7 +42,7 @@ - (instancetype)init {
 + (instancetype)sharedInstance {
```
Comment (MehrdadMzfr) [inline MobileCenterPush/MobileCenterPush/MSPush.m]: ?

File: MobileCenterPush/MobileCenterPush/MSPush.m [hunk 4]
```diff
@@ -82,6 +82,7 @@ + (void)didFailToRegisterForRemoteNotificationsWith:(NSError *)error {
 #pragma mark - MSServiceAbstract
 
 - (void)applyEnabledState:(BOOL)isEnabled {
+
```
Comment (MehrdadMzfr) [inline MobileCenterPush/MobileCenterPush/MSPush.m]: resolved

File: MobileCenterPush/MobileCenterPush/MSPush.m (hunk 1 above)
Comment (MehrdadMzfr) [inline MobileCenterPush/MobileCenterPush/MSPush.m]: resolved

Comment (MehrdadMzfr) [inline MobileCenterPush/MobileCenterPush/MSPush.m]: removed

Comment (MehrdadMzfr) [inline MobileCenterPush/MobileCenterPush/MSPush.m]: removed

Comment (MehrdadMzfr) [inline MobileCenterPush/MobileCenterPush/MSPush.m]: removed

File: MobileCenterPush/MobileCenterPush/MSPush.m [hunk 5]
```diff
@@ -102,50 +103,45 @@ + (void)resetSharedInstance {
...
 }
 
 #ifdef __IPHONE_8_0
-- (void)application:(UIApplication *)application
-    didRegisterUserNotificationSettings:(UIUserNotificationSettings *)notificationSettings {
-  (void)notificationSettings;
-
-  // register to receive notifications
+- (void)application:(UIApplication *)application didRegisterUserNotificationSettings:(UIUserNotificationSettings *)notificationSettings
+{
```
Comment (MehrdadMzfr) [inline MobileCenterPush/MobileCenterPush/MSPush.m]: resolved

**Comment by jaelim-ms** [comment]
Merging this branch to a new feature branch. Will be merged into base/push later.

**Pull request merged**